fuzzamoto-scenarios/bin/ir.rs

```diff
@@ -200,17 +213,30 @@ where
                     if self.inner.connections.is_empty() {
                         return;
                     }
-
                     let num_connections = self.inner.connections.len();
-                    if let Some(connection) = self.inner.connections.get_mut(from % num_connections)
-                    {
-                        if cfg!(feature = "force_send_and_ping") {
+                    let dst = from % num_connections;
+
+                    if let Some(connection) = self.inner.connections.get_mut(dst) {
+                        if self.logging {
+                            if let Ok(received) = connection.send_and_recv(&(command, message)) {
+                                log::info!("We've received {:?}", received);
+                                self.received.extend(received);
+                            }
+                        } else if cfg!(feature = "force_send_and_ping") {
                             let _ = connection.send_and_ping(&(command, message));
```

Comment:
```suggestion
                            let _ = connection.send_and_ping(&(command, message), self.logging);
```

I think it might be cleaner to add a boolean as parameter to `send_and_ping` to decide whether or not to record. Otherwise we duplicate `send_and_ping` and `wait_for_pong`.